### PR TITLE
FYLE-2dmkmkj: Moved per diem service to platform and added per diem models

### PR DESCRIPTION
**Pull request by Dimple16**

### CODE REVIEW

**Review by rvab — APPROVED** [review]

File: src/app/core/services/per-diem.service.ts [hunk 1]
```diff
@@ -1,24 +1,85 @@
...
+  constructor(private spenderPlatformApiService: SpenderPlatformApiService) {}
 
   @Cacheable({
     cacheBusterObserver: perDiemsCacheBuster$,
   })
-  getRates() {
-    return this.apiService.get('/per_diem_rates');
+  getRates(): Observable<PerDiemRates[]> {
+    return this.getActivePerDiemRatesCount().pipe(
+      switchMap((count) => {
```
[the remark this answers is not in the document]
Comment (rvab) [reply src/app/core/services/per-diem.service.ts:23]: I thought to suggest the same but felt that it was too early to take that call. @Dimple16 Check if we can create a new method `getPaginatedData` and include this code in the platform service itself. Do this only if the code is duplicating.

File: src/app/core/models/platform/platform-per-diem-rates.model.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export interface PlatformPerDiemRates {
```
[the remark this answers is not in the document]
Comment (rvab) [reply src/app/core/models/platform/platform-per-diem-rates.model.ts:1]: Are we not going to remove `Class1` once all the APIs are migrated to the platform?. IMO, if we keep the same name, we need an extra effort to figure out whether the interface is for platform API or older API

File: src/app/core/services/per-diem.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/core/services/per-diem.service.ts:23]: I will move this to service in a separate pr @abhishek1234321 @rvab

File: src/app/core/models/platform/platform-per-diem-rates.model.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  id: number;
+  org_id: string;
+  created_at: Date;
+  updated_at: Date;
+  is_enabled: boolean;
+  name: string;
+  code?: string;
```
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/core/models/platform/platform-per-diem-rates.model.ts:8]: code and description are newly added columns in the platform side, we don't have them in the public

File: src/app/core/models/platform/platform-per-diem-rates.model.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/core/models/platform/platform-per-diem-rates.model.ts:1]: we will be removing `Class1` once all the migration is done and stable

**Pull request merged**